stdlib/LinearAlgebra/test/diagonal.jl

```diff
@@ -462,6 +462,26 @@ end
     @test kron(Ad, Ad).diag == kron([1, 2, 3], [1, 2, 3])
 end
 
+@testset "kron (issue #46456)" begin
+    A = Diagonal(randn(10))
+    BL = Bidiagonal(randn(10), randn(9), :L)
+    BU = Bidiagonal(randn(10), randn(9), :U)
+    C = SymTridiagonal(randn(10), randn(9))
+    D = Tridiagonal(randn(9), randn(10), randn(9))
+    ABL = kron(A, BL)
+    ABU = kron(A, BU)
+    AC = kron(A, C)
+    AD = kron(A, D)
+    @test ABL isa Bidiagonal
+    @test ABU isa Bidiagonal
+    @test AC isa SymTridiagonal
+    @test AD isa Tridiagonal
+    @test ABL == kron(Array(A), Array(BL))
+    @test ABU == kron(Array(A), Array(BU))
+    @test AC == kron(Array(A), Array(C))
+    @test AD == kron(Array(A), Array(D))
```

Comment:
Not essential, but shorter
```suggestion
    @test ABL::Bidiagonal == kron(Array(A), Array(BL))
    @test ABU::Bidiagonal == kron(Array(A), Array(BU))
    @test AC::SymTridiagonal == kron(Array(A), Array(C))
    @test AD::Tridiagonal == kron(Array(A), Array(D))
```

Reply:
Will do.

Reply:
Done.